**Project Title**: Student Marks Analyzer


**Problem Statement:** Develop a program that analyzes student marks using Python and NumPy.


**Logic Explaination:** The program starts by taking the number of students as input and then collects their marks using a loop. These marks are stored in a list and converted into a NumPy array. It then calculates the total, average, highest, and lowest marks using NumPy functions. Based on the average, a grade is assigned using if-else conditions. Finally, all the results are displayed and the program ends.

In [4]:
import numpy as np

def get_marks():
    n = int(input("Enter number of students: "))
    marks = []

    for i in range(n):
        m = float(input(f"Enter marks for student {i+1}: "))
        marks.append(m)

    return np.array(marks)


def calculate_results(marks):
    total = np.sum(marks)
    average = np.mean(marks)
    highest = np.max(marks)
    lowest = np.min(marks)

    return total, average, highest, lowest


def assign_grade(avg):
    if avg > 85:
        return "A"
    elif avg >= 70:
        return "B"
    elif avg >= 50:
        return "C"
    else:
        return "Fail"


def display_results(total, avg, high, low, grade):
    print("\n===== RESULT =====")
    print("Total Marks:", total)
    print("Average Marks:", avg)
    print("Highest Marks:", high)
    print("Lowest Marks:", low)
    print("Grade:", grade)


# Main Program
marks = get_marks()
total, avg, high, low = calculate_results(marks)
grade = assign_grade(avg)

display_results(total, avg, high, low, grade)

Enter number of students: 3
Enter marks for student 1: 80
Enter marks for student 2: 65
Enter marks for student 3: 78

===== RESULT =====
Total Marks: 223.0
Average Marks: 74.33333333333333
Highest Marks: 80.0
Lowest Marks: 65.0
Grade: B


#Sample Output
Enter number of students: 3
Enter marks for student 1: 80
Enter marks for student 2: 65
Enter marks for student 3: 78

===== RESULT =====
Total Marks: 223.0
Average Marks: 74.33333333333333
Highest Marks: 80.0
Lowest Marks: 65.0
Grade: B



